synth.py

```diff
@@ -129,19 +129,25 @@ def client_for_repo(repo_slug) -> Optional[CloudClient]:
 
 def allowed_repo(repo) -> bool:
     return (
-        repo["language"] == "python"
-        and repo["repo"].startswith("googleapis/python-")
-        and repo["repo"] not in REPO_EXCLUSION
+        repo["full_name"].startswith("googleapis/python-")
+        and repo["full_name"] not in REPO_EXCLUSION
+        and not repo["archived"]
     )
 
 
+def get_clients_batch_from_response_json(response_json) -> List[CloudClient]:
+    return [client_for_repo(repo["full_name"]) for repo in response_json if allowed_repo(repo)]
+
 def all_clients() -> List[CloudClient]:
-    response = requests.get(REPO_LIST_JSON)
-    clients = [
-        client_for_repo(repo["repo"])
-        for repo in response.json()["repos"]
-        if allowed_repo(repo)
-    ]
+    clients = []
+    page_number = 1
+    while page_number < 10:
```

Comment:
Can the link header be used to terminate the loop instead? https://2.python-requests.org/en/master/user/advanced/#id14

Reply:
Done.